### PR TITLE
Make font store has more flexibility to accept different type of glyph store

### PR DESCRIPTION
**Pull request by andy840119**
Make `font store` has more flexibility to accept different types of `GlyphStore`.
For letting users customize glyph stores like `TtfGlyphStore` or `OtfGlyphStore`.

And here's the sample on rendering customize `.ttf` font in osu-framework:
- https://github.com/andy840119/osu-framework/pull/2

### CODE REVIEW

File: osu.Framework/IO/Stores/FontStore.cs [hunk 1]
```diff
@@ -194,11 +195,12 @@ public ITexturedCharacterGlyph Get(string fontName, char character)
...
+            var matchedGlyphStore = glyphStores.FirstOrDefault(x => x.FontName == fontName);
+
+            if (matchedGlyphStore != null)
             {
-                var bh = store.GetBaseHeight(fontName);
-                if (bh.HasValue)
-                    return bh.Value / ScaleAdjust;
+                var bh = matchedGlyphStore.GetBaseHeight();
+                return bh / ScaleAdjust;
             }
```
Comment (frenzibyte) [inline osu.Framework/IO/Stores/FontStore.cs:204]: Not 100% sure about switching to LINQ on such method, could just add a `continue` until the font name matches.

**Review by frenzibyte — CHANGES_REQUESTED** [review]

**Comment by frenzibyte** [comment]
Please refrain from force-pushing here.

File: osu.Framework/IO/Stores/IGlyphStore.cs [hunk 2]
```diff
@@ -10,6 +11,27 @@ namespace osu.Framework.IO.Stores
...
+        /// </summary>
+        /// <param name="c">The character.</param>
+        /// <returns>Has glyph.</returns>
+        bool HasGlyph(char c);
+
+        /// <summary>
+        /// Get base height.
+        /// </summary>
+        /// <returns>The height.</returns>
+        int GetBaseHeight();
```
[the remark this answers is not in the document]
Comment (frenzibyte) [reply osu.Framework/IO/Stores/IGlyphStore.cs]: It should be removed if the method itself is written only to return something. I would even change it to a property but that's out of scope I guess.

**Review by frenzibyte — APPROVED** [review]
Was hoping for a proper abstraction and *potentially* reuse of `RawCachingGlyphStore` for custom glyph stores, but not going to insist too much on that with the current state it's in.

**Review by bdach — APPROVED** [review]
I guess I don't see any harm with this if this is enough exposition.

**Review by peppy — APPROVED** [review]

**Pull request merged**